e2e-polybft/e2e/bridge_test.go

```diff
@@ -287,14 +285,164 @@ func TestE2E_Bridge_Transfers(t *testing.T) {
 
 		// the transactions are mined and state syncs should be executed by the relayer
 		// and there should be a success events
-		var stateSyncedResult contractsapi.StateSyncResultEvent
-
 		logs, err := getFilteredLogs(stateSyncedResult.Sig(), initialBlockNum, finalBlockNum, childEthEndpoint)
 		require.NoError(t, err)
 
 		// assert that all state syncs are executed successfully
 		checkStateSyncResultLogs(t, logs, transfersCount)
 	})
+
+	t.Run("non native ERC20 deposit and withdraw", func(t *testing.T) {
+		rootchainTxRelayer, err := txrelayer.NewTxRelayer(txrelayer.WithIPAddress(cluster.Bridge.JSONRPCAddr()))
+		require.NoError(t, err)
+
+		childchainTxRelayer, err := txrelayer.NewTxRelayer(txrelayer.WithClient(validatorSrv.JSONRPC()))
+		require.NoError(t, err)
+
+		// Deploy ERC20 contract to the rootchain
+		rootchainDeployer, err := rootHelper.DecodePrivateKey("")
+		require.NoError(t, err)
+
+		txn := &ethgo.Transaction{To: nil, Input: contractsapi.RootERC20.Bytecode}
+		receipt, err := rootchainTxRelayer.SendTransaction(txn, rootchainDeployer)
+		require.NoError(t, err)
+		require.NotNil(t, receipt)
+
+		contractAddress := receipt.ContractAddress
+
+		t.Log("Rootchain token address:", contractAddress)
+
+		// wait for next sprint block as the starting point,
+		// in order to be able to make assertions against blocks offsetted by sprints
+		initialBlockNum, err := childEthEndpoint.BlockNumber()
+		require.NoError(t, err)
+
+		initialBlockNum = initialBlockNum + sprintSize - (initialBlockNum % sprintSize)
+		require.NoError(t, cluster.WaitForBlock(initialBlockNum, 1*time.Minute))
+
+		t.Log("Initial block number:", initialBlockNum)
+
+		// get initial exit id
+		initialExitEventID := getLastExitEventID(t, childchainTxRelayer)
+
+		// send a few transactions to the bridge
+		require.NoError(t, cluster.Bridge.Deposit(
+			common.ERC20,
+			types.Address(contractAddress),
+			polybftCfg.Bridge.RootERC20PredicateAddr,
+			rootHelper.TestAccountPrivKey,
+			strings.Join(receivers[:], ","),
+			strings.Join(amounts[:], ","),
+			"",
+			cluster.Bridge.JSONRPCAddr(),
+			rootHelper.TestAccountPrivKey,
+			false,
+		))
+
+		// wait for a few more sprints
+		finalBlockNumber := initialBlockNum + 10*sprintSize
+		require.NoError(t, cluster.WaitForBlock(finalBlockNumber, 2*time.Minute))
+
+		t.Log("Final block number:", finalBlockNumber)
+
+		// the transactions are processed and there should be a success events
+		logs, err := getFilteredLogs(stateSyncedResult.Sig(), initialBlockNum, finalBlockNumber, childEthEndpoint)
+		require.NoError(t, err)
+
+		// assert that all deposits are executed successfully
+		// deposit transaction only
```

Comment:
Consider either deleting this line or updating it (something like this):
```suggestion
		// (map token action is executed along with the first deposit transaction)
```

Reply:
Done